cli.py

```diff
@@ -31,12 +31,12 @@ def print_import_errs():
     for ie in import_errs:
         util.verbose(ie)
 
-if 'COLUMNS' not in os.environ:
-    # Create a custom width formatter by fixing two arguments for the default formatter class, namely
-    # 'width' (defaults to 80 - 2) and 'max_help_position' (defaults to 24)
-    FORMATTER_CLASS = functools.partial(argparse.HelpFormatter, max_help_position=40, width=100)
-else:
-    FORMATTER_CLASS = argparse.HelpFormatter
+column_width = ctx.get_config('command-line.output_columns') or os.environ.get('COLUMNS') or 80
+max_help_position = max(column_width - 56, 24) # TODO: Too arbitrary
```

Comment:
please change the configuration logic in such a way that it is checked whether _any_ specific configuration is found (either via env or from cfg_file); if not, do "nothing" (use argparse's default behaviour)